### PR TITLE
Add handling of remaining semantic types

### LINKED ISSUE #58: Update CACE to deal with table template type

### PR DESCRIPTION
**Pull request by BrianSipos**
This refactors other semantic types for consistency.

Closes #58 and #71

### CODE REVIEW

**Comment by BrianSipos** [comment]
@justinethier there are some base unit test cases but if you'd like to add others or make the semtype test functions more consistent that is a welcome addition.

File: src/cace/amm/semtype.h [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+ */
+int amm_type_set_use_direct(amm_type_t *type, const amm_type_t *base);
+
+/** A closed interval of size_t values with optional minimum and maximum.
+ */
+typedef struct
+{
+    /// True if this interval has a finite minimum
+    bool has_min;
+    /// The minimum value of the interval, valid if #has_max is true
```
Comment (justinethier) [inline src/cace/amm/semtype.h:79]: **Nitpick**: Change to `has_min` in the comment

File: src/cace/amm/typing.h [hunk 2]
```diff
@@ -70,7 +71,7 @@ void amm_type_deinit(amm_type_t *type);
  */
 void amm_type_reset(amm_type_t *type);
 
-/// OPLIST for the amm_type_t
+/// M*LIB OPLIST for the amm_type_t
```
Comment (justinethier) [inline src/cace/amm/typing.h:74]: **Praise**: Its helpful to clarify this, thanks!

File: test/cace/test_amm_typing.c [hunk 3]
```diff
@@ -155,6 +156,155 @@ void test_amm_type_match_semtype_use_1(const char *inhex, bool expect)
...
+
+    check_match(&mytype, inhex, expect);
+    amm_type_deinit(&mytype);
+}
+
+TEST_CASE("F6", false)          // ari:null
+TEST_CASE("8211820102", false)  // ari:/AC/(1,2)
+TEST_CASE("82118201F5", false)  // ari:/AC/(1,true)
+TEST_CASE("82138102", true)     // ari:/TBL/c=3;
+TEST_CASE("8213830201F5", true) // ari:/TBL/c=3;(1,2,3)
```
Comment (justinethier) [inline test/cace/test_amm_typing.c:284]: **Issue**: This comment is wrong, should be `TBL/c=2;(1,true)`

**Review by justinethier — APPROVED** [review]
PR is good to go!
I added a few more tests cases for consistency, feel free to review prior to merge.

**Pull request merged**